### PR TITLE
Add inlay hint support

### PR DESCRIPTION
**Pull request by kitsonk**
Ref #956

### CODE REVIEW

**Comment by matklad** [comment]
dupe of https://github.com/microsoft/language-server-protocol/pull/1249 ?

File: _specifications/specification-3-17.md [hunk 1]
```diff
@@ -8915,11 +8930,125 @@ Servers usually support different communication channels (e.g. stdio, pipes, ...
...
+
+	/**
+	 * Whitespace before the hint.
+	 */
+	whitespaceBefore?: boolean;
+
+	/**
+	 * Whitespace after the hint.
+	 */
+	whitespaceAfter?: boolean;
```
Comment (rcjsuen) [inline _specifications/specification-3-17.md:9040]: Can we include a note in the comment that states what happens when these optional fields are omitted?

**Comment by kitsonk** [comment]
@matklad that hasn't been aligned to what vscode implemented and hasn't been updated since 8th June, which this PR is aligned to and I raised it based off this comment https://github.com/microsoft/language-server-protocol/issues/956#issuecomment-912357829.

I would be fine with #1249 being aligned to what shipped in vscode.  All I want to do is get the feature added, by hook or crook.

**Comment by kitsonk** [comment]
Also @matklad this comment seems to indicate that the PR is "obsolete" but left up to @dbaeumer to close when appropriate: https://github.com/microsoft/language-server-protocol/pull/1249#issuecomment-854767339

**Comment by dbaeumer** [comment]
Yes, I think we should continue in #1249

**Pull request closed without merging**